check.js

```diff
@@ -8,29 +8,35 @@ function auditOk(issues) {
 }
 
 function printHumanReadableReport(issues) {
-    if (auditOk(issues)) {
-        console.log("audit ok.");
-        return;
-    }
-    issues.forEach(issue => {
+    return new Promise((resolve) => {
+        if (auditOk(issues)) {
+            console.log("audit ok.");
+            return resolve(issues);
+        }
+        issues.forEach(issue => {
+            console.log(
+                `--------------------------------------------------`
+            );
+            console.log(`[${issue.severity}] ${issue.title}`);
+            console.log(issue.items.map(item => item.report).join("\n"));
+        });
+
+        console.log(
+            `--------------------------------------------------`
+        );
+        console.error(" 😱   Unresolved issues found!");
         console.log(
             `--------------------------------------------------`
         );
-        console.log(`[${issue.severity}] ${issue.title}`);
-        console.log(issue.items.map(item => item.report).join("\n"));
-    });
 
-    console.log(
-        `--------------------------------------------------`
-    );
-    console.error(" 😱   Unresolved issues found!");
-    console.log(
-        `--------------------------------------------------`
-    );
+        resolve(issues);
+    });
 }
 
 function printJsonReport(issues) {
-    console.log(JSON.stringify(issues));
+    return new Promise((resolve) => {
+        process.stdout.write(JSON.stringify(issues), () => resolve(issues));
```

Comment:
This one is what I'd like to use everywhere. might wrap it a bit different

Reply:
oh well, I tried it in v2 and would have to rewrite and destabilize too much to make all printing asynchronous.

Reply:
```
const data = '{TEST TEST}'.repeat(1024 * 8);
if (process.stdout._handle) {
	process.stdout._handle.setBlocking(true);
}
console.log(data);
process.exit(0);
```

will also force it to wait for a buffer flush - but I'm not sure if that gives any other bad side effects.

Reply:
I went with async solution (in refactoring) for now, but making everything consistent and synchronous is also tempting.